### PR TITLE
feat(react-router): add react-router to app shell

### PR DESCRIPTION
**Pull request by DanBeckDev**

### CODE REVIEW

File: apps/MFE/Footer/package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "@batman-ui-components/text": "1.0.0",
     "@batman/core-logger": "*",
     "@batman/federate-component": "*",
+    "@batman/header": "^0.1.1",
```
Comment (velvetkevorkian) [inline apps/MFE/Footer/package.json:23]: Is this right? I thought we federated to avoid this sort of thing?

File: apps/MFE/Header/package.json [hunk 2]
```diff
@@ -20,6 +20,7 @@
     "@batman-ui-components/text": "1.0.0",
     "@batman/core-logger": "*",
     "@batman/federate-component": "*",
+    "@batman/header": "^0.1.1",
```
Comment (velvetkevorkian) [inline apps/MFE/Header/package.json:23]: the header depends on itself?

File: apps/MFE/ProductListing/src/react-app-env.d.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
+/* eslint-disable @typescript-eslint/triple-slash-reference */
```
Comment (velvetkevorkian) [inline apps/MFE/ProductListing/src/react-app-env.d.ts:1]: probably not for this PR, but should this be a monorepo-level config?

File: libs/framework/middlewares/src/prerender.tsx [hunk 4]
```diff
@@ -44,7 +51,14 @@ export const prerenderMiddleware = remoteEntry => {
 
       let timeout
 
-      const el = React.createElement(Component, props || {}, `\u200Cchildren\u200C`)
+      // RootContextProvider . Do we need it? What to supply as context?
+      // RoutePendingContextPorivder . Do we need it?
+      // StaticRouter . Do we need it? Supply the correct location through the request somehow
```
Comment (velvetkevorkian) [inline libs/framework/middlewares/src/prerender.tsx:56]: ?

**Review by velvetkevorkian — COMMENTED** [review]
Is there a reason a lot of MFEs have added `@batman/header` to their dependencies?

**Review by AlzAmido — APPROVED** [review]

**Pull request merged**